Sources/SwiftParser/IncrementalParseTransition.swift

```diff
@@ -57,36 +61,57 @@ public final class IncrementalParseTransition {
   ///   - reusedNodeDelegate: Optional delegate to accept information about the
   ///                         reused regions and nodes.
   public init(
-    previousTree: SourceFileSyntax,
-    edits: ConcurrentEdits,
+    previousTree: SourceFileSyntax? = nil,
+    edits: ConcurrentEdits? = nil,
     reusedNodeDelegate: IncrementalParseReusedNodeDelegate? = nil
   ) {
     self.previousTree = previousTree
     self.edits = edits
     self.reusedDelegate = reusedNodeDelegate
   }
+
+  public func setupTransition(tree: SourceFileSyntax, edits: ConcurrentEdits, delegate: IncrementalParseReusedNodeDelegate? = nil) {
+    self.previousTree = tree
+    self.edits = edits
+    self.reusedDelegate = delegate
+  }
+  
+  @_spi(RawSyntax)
+  public func registerNodeForIncrementalParse(offset: Int,  node: RawSyntax) {
+    
+    guard let furthestOffset,
+          furthestOffset > offset else { return }
+    
+    self.lookaheadBytes[node.id] = max(furthestOffset - offset, node.byteLength)
+  }
+
+  public func isValidTransition() -> Bool {
+    return previousTree != nil
+  }
 }
 
 /// Provides a mechanism for the parser to skip regions of an incrementally
 /// updated source that was already parsed during a previous parse invocation.
 public struct IncrementalParseLookup {
   fileprivate let transition: IncrementalParseTransition
-  fileprivate var cursor: SyntaxCursor
```

Comment:
As discussed offline: Is it faster to have `cursor` as a member of `IncrementalParseLookup` or always generate it in `cursorLookup`. I would expect the former to be faster, but I might be wrong.

Reply:
I think making `cursor` a member of `IncrementalParseLookup` is a little bit faster than generating it in `cursorLookup`.